keps/sig-storage/2268-non-graceful-shutdown/README.md

```diff
@@ -0,0 +1,404 @@
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Alternatives](#alternatives)
+  - [rely on the taint as a lock mechanism (taint and untaint by the same actor)](#rely-on-the-taint-as-a-lock-mechanism-taint-and-untaint-by-the-same-actor)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+- [] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+**Note:** Any PRs to move a KEP to `implementable` or significant changes once it is marked `implementable` should be approved by each of the KEP approvers. If any of those
+approvers is no longer appropriate than changes to that list should be approved by the remaining approvers and/or the owning SIG (or SIG-arch for cross cutting KEPs).
+
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://github.com/kubernetes/enhancements/issues
+[kubernetes/kubernetes]: https://github.com/kubernetes/kubernetes
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+In case of an instance being shutdown (i.e. not running) the control plane doesn't make the right
+assumptions to enable stateful workload to fail-over, This KEP introduces a flow to do so.
+
+This KEP is part of collection of KEPs that intends to enhance stateful workload management on top of Kubernetes.
+
+## Motivation
+
+Today kubernetes doesn't enable providers that are able to safely detach (i.e. only detach when volumes aren't being written to) to do so. This proposal aims to introduce a new CSI capability that if advertised would trigger volume detach if a node is partitioned from the control plane.
+
+### Goals
+
+* Increase the availability of stateful workloads
+* Automate self-healing for stateful workloads
```

Comment:
Another goal is to target only CSI volumes. A non-goal is the existing in-tree volumes.

Reply:
Done